libraries/AP_Math/AP_Math.h

```diff
@@ -25,273 +17,115 @@
-
-// Converts from WGS84 geodetic coordinates (lat, lon, height)
-// into WGS84 Earth Centered, Earth Fixed (ECEF) coordinates
-// (X, Y, Z)
-void wgsllh2ecef(const Vector3d &llh, Vector3d &ecef);
-
-// Converts from WGS84 Earth Centered, Earth Fixed (ECEF)
-// coordinates (X, Y, Z), into WHS84 geodetic
-// coordinates (lat, lon, height)
-void wgsecef2llh(const Vector3d &ecef, Vector3d &llh);
-
-// constrain a value
-// constrain a value
-static inline float constrain_float(float amt, float low, float high)
-{
-	// the check for NaN as a float prevents propogation of
-	// floating point errors through any function that uses
-	// constrain_float(). The normal float semantics already handle -Inf
-	// and +Inf
-	if (isnan(amt)) {
-		return (low+high)*0.5f;
-	}
-	return ((amt)<(low)?(low):((amt)>(high)?(high):(amt)));
-}
-// constrain a int16_t value
-static inline int16_t constrain_int16(int16_t amt, int16_t low, int16_t high) {
-	return ((amt)<(low)?(low):((amt)>(high)?(high):(amt)));
-}
-
-// constrain a int32_t value
-static inline int32_t constrain_int32(int32_t amt, int32_t low, int32_t high) {
-	return ((amt)<(low)?(low):((amt)>(high)?(high):(amt)));
-}
-
-//matrix algebra
-bool inverse(float x[], float y[], uint16_t dim);
-
-// degrees -> radians
-static inline float radians(float deg) {
-	return deg * DEG_TO_RAD;
-}
-
-// radians -> degrees
-static inline float degrees(float rad) {
-	return rad * RAD_TO_DEG;
+template <class T>
+auto wrap_180_cd(const T &angle) -> decltype(wrap_180(angle, 100.f)) {
+    return wrap_180(angle, 100.f);
 }
 
-// square
-static inline float sq(float v) {
-	return v*v;
-}
-
-// 2D vector length
-static inline float pythagorous2(float a, float b) {
-	return sqrtf(sq(a)+sq(b));
-}
-
-// 3D vector length
-static inline float pythagorous3(float a, float b, float c) {
-	return sqrtf(sq(a)+sq(b)+sq(c));
-}
-
-#ifdef radians
-#error "Build is including Arduino base headers"
-#endif
-
+/* 
+ * @brief: Gets two values and returns the smaller one.
+ */
 template<typename A, typename B>
-static inline auto MIN(const A &one, const B &two) -> decltype(one < two ? one : two) {
+auto MIN(const A &one, const B &two) -> decltype(one < two ? one : two) { 
     return one < two ? one : two;
 }
 
+/* 
+ * @brief: Gets two values and returns the bigger one.
+ */
 template<typename A, typename B>
-static inline auto MAX(const A &one, const B &two) -> decltype(one > two ? one : two) {
+auto MAX(const A &one, const B &two) -> decltype(one > two ? one : two) {  
     return one > two ? one : two;
 }
 
-#define NSEC_PER_SEC 1000000000ULL
-#define NSEC_PER_USEC 1000ULL
-#define USEC_PER_SEC 1000000ULL
-
-inline uint32_t hz_to_nsec(uint32_t freq)
-{
-    return NSEC_PER_SEC / freq;
+/*
+ * Converter functions
+ *  - Avoid zero divisions
+ *  - Inheritss a float cast (because of PX4)
+ */
+template<class T>
+T hz_to_nsec(const T &freq) {
+    T val = !is_zero(static_cast<float>(freq)) ? NSEC_PER_SEC / freq : 0;
```

Comment:
As I said previously, I don't want floating point math when it was supposed not to be floating point. So, could you check T is floating point instead of casting?